components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/utils/ServiceCatalogUtils.java

```diff
@@ -85,17 +87,31 @@
     private static String updateServiceUrl(String currentUrl) throws ResolverException {
         /*
             Supported formats
-            https://{host}:{port}/api1
-            https://{url}/api1
+            https://{MI_HOST}:{MI_PORT}/api1
+            https://{MI_URL}/api1
         */
-        SystemResolver resolver = new SystemResolver();
-        if (currentUrl.contains(HOST) && currentUrl.contains(PORT)) {
-            resolver.setVariable(MI_HOST);
-            String host = resolver.resolve();
-            resolver.setVariable(MI_PORT);
-            String port = resolver.resolve();
-            currentUrl = currentUrl.replace(HOST, host).replace(PORT, port);
+        if (currentUrl.contains(HOST) || currentUrl.contains(PORT)) {
+            Map<String, Object> configParser = ConfigParser.getParsedConfigs();
+            String envHost = System.getenv(MI_HOST);
+            // update the {MI_HOST} with server.hostname value
+            if (currentUrl.contains(HOST) && (envHost == null || envHost.isEmpty())) {
+                envHost = (String) configParser.get(SERVER_HOSTNAME);
+            }
+            String envPort = System.getenv(MI_PORT);
+            // update the {MI_PORT} with server listener ports + offset values
+            if (currentUrl.contains(PORT) && (envPort == null || envPort.isEmpty())) {
+                int portOffset = Integer.parseInt(System.getProperty(SERVER_PORT_OFFSET));
+                if (currentUrl.startsWith("https")) {
+                    int httpsPort = Integer.parseInt((String) configParser.get(HTTPS_LISTENER_PORT));
```

Comment:
Let's use the existing ways of reading the value from the axis2.xml

Reply:
Added Utils methods to get Hostname, HTTP port and HTTPS port.